src/actions/manager.tsx

```diff
@@ -97,12 +98,31 @@ export class ActionManager implements ActionsManagerInterface {
     );
   }
 
+  showAllowAction = (name: ActionName) => {
+    const canvasActions = this.getAppState().canvasActions;
+
+    if (canvasActions) {
+      switch (name) {
+        case "saveScene":
+        case "saveAsScene":
+        case "clearCanvas":
+          return canvasActions[name];
+      }
+    }
+
+    return true;
+  };
```

Comment:
This function is not needed, you can use the  value of `props.canvasAction[action.name]` to decide whether to show the  prop.

Reply:
That results in "Element implicitly has an 'any' type because expression of type 'ActionName' can't be used to index type 'CanvasActions'."

Won't doing that will allow the user to pass any action name as an attribute and disable it?

Reply:
For that I would suggest to whitelist before passing it to `App.tsx` so you can just use  `props.UIOptions.canvasAction[action.name]` instead of calling this function for every action.

Reply:
It was causing the same problem. Passing the key as `action.name as keyof typeof canvasActions` fixed it. I removed the function now.